en/simulation/hitl.md

```diff
@@ -1,101 +1,133 @@
+   
+Once configuration is complete, **close** *QGroundControl* and disconnect the flight controller hardware from the computer. 
 
+### Simulator Setup
+
+Follow the appropriate setup steps for your simulator in the following sections.
+
+#### Gazebo
+
+> **Note** Make sure *QGroundControl* is not running!
+
+1. Update the environment variables: 
+  ```sh
+  cd <Firmware_clone>
+  make posix_sitl_default gazebo
+  source Tools/setup_gazebo.bash $(pwd) $(pwd)/build/posix_sitl_default
   ```
-  ./Tools/jmavsim_run.sh -q -d /dev/ttyACM0 -b 921600 -r 250
+1. Open the vehicle model's sdf file (e.g. **Tools/sitl_gazebo/models/iris/iris.sdf**).
+1. Under the `mavlink_interface plugin` section, change the `serialEnabled` and `hil_mode` parameters to `true`. 
+
+  ![HIL Parameters](../../assets/simulation/gazebo_sdf_model_hil_params.png)
+1. Replace the `serialDevice` parameter (`/dev/ttyACM0`) if necessary.
+1. Connect the flight controller to the computer and wait for it to boot.
+1. Run Gazebo in HITL mode 
+  ```sh
+  gazebo /Tools/sitl_gazebo/worlds/iris.world
   ```
+1. Start *QGroundControl*. It should autoconnect to PX4 and Gazebo.
 
-* Then run QGroundControl - it will auto-connect.
+#### jMAVSim (Quadrotor only)
 
-### Using X-Plane and QGroundControl \(Fixed Wing only\)
+> **Note** Make sure *QGroundControl* is not running!
 
-In X-Plane two key settings have to be made: In Settings -&gt; Data Input and Output, set these checkboxes:
+1. Run jMAVSim in HITL mode (replace the serial port name `/dev/ttyACM0` if necessary - e.g. on Mac OS this would be `/dev/tty.usbmodem1`):
+  ```sh
+  ./Tools/jmavsim_run.sh -q -d /dev/ttyACM0 -b 921600 -r 250
+  ```
+1. Start *QGroundControl*. It should autoconnect to PX4 and jMAVSim.
 
-![](../../assets/gcs/xplane_data_config.png)
+#### Using X-Plane (Fixed Wing only)
 
-In Settings -&gt; Net Connections in the Data tab, set localhost and port 49005 as IP address, as shown in the screenshot below:
+1. Open X-Plane
+1. In **Settings > Data Input and Output**, set these checkboxes:
 
-![](../../assets/gcs/xplane_net_config.png)
+   ![X-Plane data configuration](../../assets/gcs/xplane_data_config.png)
 
-#### Enable HITL in QGroundControl
+1. In **Settings > Net Connections**, in the *Data* tab, set localhost and port 49005 as IP address, as shown below:
 
-Widgets -&gt; HIL Config, then select X-Plane 10 in the drop-down and hit connect. Once the system is connected, battery status, GPS status and aircraft position should all become valid:
+   ![X-Plane network configuration](../../assets/gcs/xplane_net_config.png)
 
-![](../../assets/gcs/qgc_sim_run.png)
+1. Enable HITL in QGroundControl
+   1. Open *QGroundControl*
+   1. Open **Widgets > HIL Config**. Select X-Plane 10 in the drop-down and hit connect. Once the system is connected, battery status, GPS status and aircraft position should all become valid:
```

Comment:
@dagar A bit annoying that X-Plane requires specific setup. Should I change this to "Enable X-Plane in QGC"?

Reply:
I'm wondering if we should not publicize xplane HIL at the moment. There are several issues, and even when it works the update rate (framerate) is way too slow to run the system realistically. Things in the vehicle running at 40-50 Hz (your game framerate) that typically run at 250 Hz.

Reply:
I'll just do it.

Reply:
@dagar  If we supported HTIL Plane on Gazebo that would be a no brainer, but at the moment this is the only simulator for HITL and plane - @arwagoner about plans to support HITL/FW?

Is anyone looking at fixing this? I don't want to make an arbitrary decision because there is a bit of work to untangle X-Plane out of this.